lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -281,7 +281,13 @@ fn generate_constraint_init_group(f: &Field, c: &ConstraintInitGroup) -> proc_ma
     let seeds_with_nonce = match &c.seeds {
         None => quote! {},
         Some(c) => {
-            let s = &c.seeds;
+            let s = &mut c.seeds.clone();
+            // If the seeds came with a trailing comma, we need to chop it off
+            // before we interpolate them below.
+            if s.trailing_punct() {
+                let last = s.pop().unwrap();
+                s.push_value(last.value().clone());
```

Comment:
This should remove extra clone.
```suggestion
                s.push_value(last.into_value());
```


Also, maybe better skip `trailing_punct()`?
```rust
if let Some(pair) = s.pop() {
    s.push_value(pair.into_value());
}
```

Reply:
Ooh, nice, I'll change that 👍